packages/roosterjs-editor-dom/lib/clipboard/createFragmentFromClipboardData.ts

```diff
@@ -106,6 +109,10 @@ export default function createFragmentFromClipboardData(
         img.style.maxWidth = '100%';
         img.src = imageDataUri;
         fragment.appendChild(img);
+        // If the fragment is pasted as image, the excel attributes are not needed anymore, so remove it to avoid treat the fragment as a excel table.
+        if (pasteAsImage) {
```

Comment:
This is so hacky. `createFragmentFromClipboardData` is a common function so it should not know the detail of paste source. We need another way to do this.

Reply:
For example, in the code convert Excel table, we check if there is only an image in the fragment.

Reply:
Or we can add a property in BeforePasteEvent to specify the paste option, so that plugins can read it and only change paste content when it is a "normal paste"

Reply:
is it safe to only change content when is "normal paste"? For now we are changing for all existing types of paste, maybe change the content only for normal paste can cause a regression.

Reply:
Maybe not only "normal paste", but also "paste and merge format".

First we need to define what paste actions are supported. For now we support these 4:
- Normal paste  (default option)
- Paste as plain text: Then we will only get plain text and do not fix up any HTML
- Paste and merge format: Similar with normal paste, but change the format
- Paste image: Only image is pasted, all other HTML is removed, so plugin does not need to fix anything


In general, plugin knows what kind of fix it need to do, for exmaple, if we paste from Word, plugin that handles Word paste can know what scenario it should fix. If it also want to fix image paste, yes it can. So from a common code we just provide the basic information and let plugin decide what to do.

Normal paste/paste as plain text/paste image/ ... these are the different options selected by user, so plugins should also know this info and do proper handling.

Reply:
I made a new change that I added a enum with the paste types then createPasteFragment function provide the paste type data to the event data. Then I only added a check if the paste type is asImage, do not convert from excel. Do you think that I should add a check that only do any conversion when is Normal and MergeFormat paste types?

Reply:
I think we just need to check Normal and MergeFormat